### PR TITLE
Fix error handling in X509_REQ_print_ex

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

File: crypto/x509/t_req.c [hunk 1]
```diff
@@ -166,13 +171,16 @@ int X509_REQ_print_ex(BIO *bp, X509_REQ *x, unsigned long nmflags,
...
-                i2a_ASN1_OBJECT(bp, obj);
+                if (i2a_ASN1_OBJECT(bp, obj) <= 0)
+                    goto err;
                 critical = X509_EXTENSION_get_critical(ex);
                 if (BIO_printf(bp, ": %s\n", critical ? "critical" : "") <= 0)
                     goto err;
                 if (!X509V3_EXT_print(bp, ex, cflag, 16)) {
-                    BIO_printf(bp, "%16s", "");
-                    ASN1_STRING_print(bp, X509_EXTENSION_get_data(ex));
+                    if (BIO_printf(bp, "%16s", "") <= 0)
```
Comment (richsalz) [inline crypto/x509/t_req.c]: I'd make this one if, but no big deal.

Comment (bernd-edlinger) [reply crypto/x509/t_req.c]: I can do this.

Comment (bernd-edlinger) [reply crypto/x509/t_req.c]: I also wonder about the best way to check the BIO_write calls here:
some test for BIO_write(bp, "string", x) <= 0 but if you write x characters, the
only good return code is x, less than x would be possible if the disk is full.

Comment (bernd-edlinger) [reply crypto/x509/t_req.c]: should I rewrite that to if (BIO_write(..., x) != x) goto err; ?

**Comment by richsalz** [comment]
Reconfirm.

**Comment by bernd-edlinger** [comment]
Merged.  Thanks!

**Pull request closed without merging**